core/jobs_test.py

```diff
@@ -85,6 +89,29 @@ def test_create_new(self):
         self.assertFalse(MockJobManagerOne.is_active(job_id))
         self.assertFalse(MockJobManagerOne.has_finished(job_id))
 
+    def test_base_job_manager_enqueue_raises_error(self):
+        with self.assertRaisesRegexp(
+            NotImplementedError,
+            'Subclasses of BaseJobManager should implement _real_enqueue().'):
+            jobs.BaseJobManager._real_enqueue(  # pylint: disable=protected-access
+                'job_id', taskqueue_services.QUEUE_NAME_DEFAULT, None)
+
+    def test_failing_jobs(self):
+
+        # Mocks GoogleCloudStorageInputReader() to fail a job.
+        _mock_input_reader = lambda _: 1 / 0
+
+        input_reader_swap = self.swap(
+            input_readers, 'GoogleCloudStorageInputReader', _mock_input_reader)
+        assert_raises_context_manager = self.assertRaises(Exception)
```

Comment:
Any reason we aren't checking for the exact message?

Reply:
There is no exception message here but there are log messages. I added tests for them, though I just added tests for the second log message as the first one is the traceback. See here:
https://github.com/oppia/oppia/blob/238f8d5abda81d42032f71bedd25329f168d1d09/core/jobs.py#L611